agent/src/softwarecontaineragent.cpp

```diff
@@ -241,23 +247,52 @@ void SoftwareContainerAgent::setContainerName(ContainerID containerID, const std
     }
 }
 
-void SoftwareContainerAgent::shutdownContainer(ContainerID containerID)
+bool SoftwareContainerAgent::shutdownContainer(ContainerID containerID)
 {
-    shutdownContainer(containerID, m_softwarecontainerWorkspace->m_containerShutdownTimeout);
+    return shutdownContainer(containerID, m_softwarecontainerWorkspace->m_containerShutdownTimeout);
 }
 
-void SoftwareContainerAgent::shutdownContainer(ContainerID containerID, unsigned int timeout)
+bool SoftwareContainerAgent::shutdownContainer(ContainerID containerID, unsigned int timeout)
 {
     profilefunction("shutdownContainerFunction");
     if (m_shutdownContainers) {
         SoftwareContainer *container = nullptr;
         if (checkContainer(containerID, container)) {
-            container->shutdown(timeout);
+            ReturnCode status = container->shutdown(timeout);
             deleteContainer(containerID);
```

Comment:
Why do we delete the container before we analyze the result of shutdown.
Also why do we not check the status of delete.